### PR TITLE
remove circular dependency between StorageCommunicationEnpoint and StorageProxy initialization

### PR DESCRIPTION
**Pull request by mmandlis**

### CODE REVIEW

File: src/runtime/storage/storage-proxy.ts [hunk 1]
```diff
@@ -30,37 +31,22 @@ import {CRDTTypeRecordToType, TypeToCRDTTypeRecord} from './storage.js';
...
-      storeProvider: StorageCommunicationEndpointProvider,
-      ttl = Ttl.infinite()) {
-    this.apiChannelId = apiChannelId;
-    this.store = storeProvider.getStorageEndpoint(
-      storeInfo,
-      this as unknown as StorageProxy<TypeToCRDTTypeRecord<CRDTTypeRecordToType<T>>>
-    ) as unknown as StorageCommunicationEndpoint<T>;
-    this.type = storeInfo.type;
+
+  // TODO(shans): remove apiChannelId once we're not constructing StorageProxy objects
```
Comment (shans) [inline src/runtime/storage/storage-proxy.ts:42]: can we remove apiChannelId now?

**Review by shans — APPROVED** [review]
Nice! Things are looking a lot cleaner.

File: src/runtime/storage/storage-proxy.ts (hunk 1 above)
Comment (mmandlis) [reply src/runtime/storage/storage-proxy.ts]: done

**Comment by mmandlis** [comment]
> Nice! Things are looking a lot cleaner.

thanks! it's going to get even better in the next PRs :)

**Pull request closed without merging**